packages/manager/apps/pci-private-network/src/pages/new/steps/LocalizationStep.tsx

```diff
@@ -47,26 +47,17 @@ export default function LocalizationStep(): JSX.Element {
   });
 
   const getMacroRegion = (regionName: string) => {
-    const localZonePattern = /^lz/i;
-    let macro: RegExpExecArray;
-    if (
-      localZonePattern.test(
-        regionName
-          .split('-')
-          ?.slice(2)
-          ?.join('-'),
-      )
-    ) {
-      macro = /\D{2,3}/.exec(
-        regionName
-          .split('-')
-          ?.slice(3)
-          ?.join('-'),
-      );
-    } else {
-      macro = /\D{2,3}/.exec(regionName);
-    }
-    return macro ? macro[0].replace('-', '').toUpperCase() : '';
+    const regionSubStrings = regionName.split('-');
+
+    const macroRegionMap = [
+      null,
+      regionSubStrings[0].split(/(\d)/)[0],
+      regionSubStrings[0],
+      regionSubStrings[2],
+      regionSubStrings[2] === 'LZ' ? regionSubStrings[3] : regionSubStrings[2],
+      regionSubStrings[3],
+    ];
+    return macroRegionMap[regionSubStrings.length] || 'Unknown_Macro_Region';
```

Comment:
Same function is repeated in multiple apps. Can you please move it to common folder and reuse?